encord/orm/dataset.py

```diff
@@ -120,14 +211,180 @@ def created_at(self, value: datetime) -> None:
         """Datetime will trim milliseconds for backwards compatibility."""
         self["created_at"] = value.strftime(DATETIME_STRING_FORMAT)
 
+    @property
+    def querier(self) -> Querier:
+        return self["querier"]
+
+    @querier.setter
+    def querier(self, new_querier: Querier) -> None:
+        self["querier"] = new_querier
+
+    @property
+    def frames_per_second(self) -> int:
+        if self.data_type != DataType.VIDEO:
+            return None
+        return self["frames_per_second"]
+
+    @frames_per_second.setter
+    def frames_per_second(self, new_frames_per_second: int) -> None:
+        self["frames_per_second"] = new_frames_per_second
+
+    @property
+    def duration(self) -> Optional[int]:
+        if self.data_type != DataType.VIDEO:
+            return None
+        return self["duration"]
+
+    @duration.setter
+    def duration(self, new_duration: int) -> None:
+        self["duration"] = new_duration
+
     @property
     def client_metadata(self) -> Optional[dict]:
         """
         Custom client metadata. This is null if it is disabled via the
         :class:`encord.orm.dataset.DatasetAccessSettings`
         """
+        if self["client_metadata"] is None:
+            if self.querier is not None:
+                res = self.querier.basic_getter(DataClientMetadata, uid=self.uid)
+                self["client_metadata"] = res.payload
+            else:
+                raise EncordException(f"Could not retrieve client metadata for DataRow with uid: {self.uid}")
         return self["client_metadata"]
 
+    @client_metadata.setter
+    def client_metadata(self, new_client_metadata: dict) -> None:
+        if self.querier is not None:
+            res = self.querier.basic_setter(
+                DataClientMetadata,
+                uid=self.uid,
+                payload={"new_client_metadata": new_client_metadata},
+            )
+            if res:
+                self["client_metadata"] = new_client_metadata
+            else:
+                raise EncordException(f"Could not update client metadata for DataRow with uid: {self.uid}")
+
+    def get_client_metadata(self):
+        """
+        This method lazily retrieves custom client metadata and cache it.
+        """
+        return self.client_metadata
+
+    @property
+    def width(self) -> int:
+        return self["width"]
+
+    @width.setter
+    def width(self, new_width: int) -> None:
```

Comment:
I don't think these should be ever updated right? Can you please only keep setters for items where updating makes sense?

Reply:
Then i will also remove existed setter for `created_at`, `data_type`, etc fields